### PR TITLE
[CM-1245 - Fix notification flow for agents

### PR DESCRIPTION
**Pull request by amntoppo**
## Issue with Notification:
1. Assignment messages are not coming 
2. Notifications are received even when chat is assigned to Bot
3. For "Notify everybody" - Agents are receiving notification for all messages, same in Android and Web, except the assignment message (edited) 

## How was the code tested?
- Compared with dashboard and tested it for Agent app - Notify Everybody and Automatic assignment feature
- Tested SDK notification flow

### CODE REVIEW

File: kommunicate/src/main/java/com/applozic/mobicomkit/broadcast/BroadcastService.java [hunk 1]
```diff
@@ -190,6 +181,26 @@ public static void sendNotificationBroadcast(Context context, Message message, i
...
+    //Notify everybody -> Send notification to all agent, along with assignment message
+    //Automatic assignment -> Send notification only to the assigned agent
+    private static boolean showNotificationToAgent(Context context, Contact contact, Channel channel, Message message) {
+        MobiComUserPreference userPreference = MobiComUserPreference.getInstance(context);
+        if (userPreference.getUserRoleType() == 3 || channel == null || contact == null) {
+            return false;
+        }
+        if (userPreference.isNotifyEverybody()) {
+            if (User.RoleType.BOT.getValue().equals(new AppContactService(context).getContactById(channel.getConversationAssignee()).getRoleType())) {
+                return (message.getMetadata() != null && message.getMetadata().containsKey(Message.BOT_ASSIGN));
```
Comment (adarshmishra) [inline kommunicate/src/main/java/com/applozic/mobicomkit/broadcast/BroadcastService.java:194]: @amntoppo this block is handling which case?

Comment (amntoppoKM) [reply kommunicate/src/main/java/com/applozic/mobicomkit/broadcast/BroadcastService.java:198]: @adarshmishra This handles the Assignment message notification -   "Assigned to Aman"

Comment (amntoppoKM) [reply kommunicate/src/main/java/com/applozic/mobicomkit/broadcast/BroadcastService.java:198]: Without the assignee role check, it shows notification even when bot is handling the conversation.

**Review by devashishmamgain — APPROVED** [review]

File: kommunicate/src/main/java/com/applozic/mobicomkit/broadcast/BroadcastService.java (hunk 1 above)
Comment (adarshmishra) [reply kommunicate/src/main/java/com/applozic/mobicomkit/broadcast/BroadcastService.java:198]: Is it similar to web?

Comment (amntoppoKM) [reply kommunicate/src/main/java/com/applozic/mobicomkit/broadcast/BroadcastService.java:198]: Yes, the flow is same as web now.

**Review by adarshmishra — APPROVED** [review]

**Comment by amntoppoKM** [comment]
Improved and moved this code to this PR: #341

**Pull request closed without merging**